hrm-service/src/contact-job/contact-job-processor.ts

```diff
@@ -15,20 +17,16 @@ export function processContactJobs() {
     processingJobs = true;
 
     return setInterval(async () => {
-      // This is the primary job processing 'loop'
-      // First we process any completed jobs. These completed jobs would be pulled from a queue in prod but we just have a cheap in memory simulation using an array here
       try {
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        const completedJobs = await processCompleteContactJobs();
+        const completedJobs = await pollAndprocessCompletedContactJobs();
 
-        // Process completed jobs
-        // console.log(completedJobs);
+        const dueContactJobs = await pullDueContactJobs(
+          subMilliseconds(new Date(), JOB_RETRY_INTERVAL_MILLISECONDS),
+        );
 
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        const pusblishResults = await publishPendingContactJobs(JOB_RETRY_INTERVAL_MILLISECONDS);
-
-        // Process published results
-        // console.log(pusblishResults);
+        const publishedContactJobResult = await publishDueContactJobs(dueContactJobs);
```

Comment:
I separated the "pull due" from the "publish due" steps to make easier the replacement of the publish piece if we want to. Also it makes easier the tests :P